cmd/kube-apiserver/app/server.go

```diff
@@ -245,6 +246,8 @@ func (s *APIServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.KubeletConfig.CAFile, "kubelet-certificate-authority", s.KubeletConfig.CAFile, "Path to a cert. file for the certificate authority.")
 	//See #14282 for details on how to test/try this option out.  TODO remove this comment once this option is tested in CI.
 	fs.IntVar(&s.KubernetesServiceNodePort, "kubernetes-service-node-port", 0, "If non-zero, the Kubernetes master service (which apiserver creates/maintains) will be of type NodePort, using this as the value of the port. If zero, the Kubernetes master service will be of type ClusterIP.")
+	// TODO: delete this flag as soon as we identify and fix all clients that send malformed updates, like #14126.
+	fs.BoolVar(&validation.FixUpdateAtServer, "repair-malformed-updates", false, "If true, server will do its best to fix the update request to pass the validation, e.g., setting empty UID in update request to its existing value.")
```

Comment:
a nit-pick: not only old kubelets, also anything that uses old client. (e.g. the scheduler also generates Events so it  needs to be recompiled as well)